### ISSUE TITLE
Query results are inconsistent within a short period of time

### ISSUE THREAD

**Issue by gsakun** [VictoriaMetrics/VictoriaMetrics#861]
**Describe the bug**
Query results are inconsistent within a short period of time
Below is a screenshot of the call API interface
![001](https://user-images.githubusercontent.com/24608247/97253046-6539db80-1846-11eb-9ead-c14254c9f105.jpg)
![002](https://user-images.githubusercontent.com/24608247/97253050-67039f00-1846-11eb-97a1-7c5ffaef4d99.jpg)
Below is a screenshot of promxy
![image](https://user-images.githubusercontent.com/24608247/97253169-b944c000-1846-11eb-9cb3-eaebf3ba96c0.png)
Below is a screenshot of the original Prometheus
![image](https://user-images.githubusercontent.com/24608247/97253200-c82b7280-1846-11eb-95f3-a9484330d769.png)

**Expected behavior**
The same data is returned each time

**Version**
```
/ # ./vmselect-prod -version
vmselect-20201013-142542-tags-v1.44.0-cluster-0-g4727bad12

/ # ./vmstorage-prod -version
vmstorage-20201013-142921-tags-v1.44.0-cluster-0-g4727bad12
```

**Used command-line flags**
```
        -search.cacheTimestampOffset=5m
        -search.denyPartialResponse
```

**Comment by starsliao** [comment]
I also found this problem in v1.44.0, but v1.43.0 has no such problem

**Comment by amravyan** [comment]
I've also faced this issue in v1.44.0 (btw, /api/v1/query_range endpoint works fine). 
Downgrading to v1.43 fixed it.

![image](https://user-images.githubusercontent.com/34545055/97581264-682efa80-1a05-11eb-8d60-717add9e869e.png)

**Comment by f41gh7** [comment]
Thanks for reporting.

Duplicates https://github.com/VictoriaMetrics/VictoriaMetrics/issues/845

Will be fixed in the next release.

**Comment by valyala** [comment]
Closing this bug as a duplicate for https://github.com/VictoriaMetrics/VictoriaMetrics/issues/845 . Note that it should be fixed in the commit 7c0b658865e6f8a6932779dbfbd23781562c8eea .
